### PR TITLE
Validate kube-proxy IPVS configuration

### LINKED ISSUE #53852: Should validate ipvs proxy options
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Some kube-proxy options are not validated.

**What you expected to happen**:

We should validate these options before running.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig network

/area kube-proxy

### PR DESCRIPTION
**Pull request by frodenas**
**What this PR does / why we need it**:

`kube-proxy` IPVS configuration was not validated when using a config file. This PR adds validation for IPVS configuration.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #53852 

**Special notes for your reviewer**:

**Release note**:

```release-note
IPVS configuration is now validated before starting `kube-proxy`
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
/cc @m1093782566
/ok-to-test
BTW, I don't think this PR needs a release note :)

**Comment by m1093782566** [comment]
a duplication of #53780?

**Comment by m1093782566** [comment]
Sorry my initial PR didn't point to my issue at first. However, I think discussing below the issue may help avoid PR duplication? FYI:)

File: cmd/kube-proxy/app/validation.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func Validate(config *componentconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateKubeProxyConntrackConfiguration(config.Conntrack, newPath.Child("KubeProxyConntrackConfiguration"))...)
 	allErrs = append(allErrs, validateProxyMode(config.Mode, newPath.Child("Mode"))...)
 	allErrs = append(allErrs, validateClientConnectionConfiguration(config.ClientConnection, newPath.Child("ClientConnection"))...)
+	allErrs = append(allErrs, validateKubeProxyIPVSConfiguration(config.IPVS, newPath.Child("KubeProxyIPVSConfiguration"))...)
```
Comment (m1093782566) [inline cmd/kube-proxy/app/validation.go]: Sorry, It almost a duplication of #53780?

**Comment by m1093782566** [comment]
Thanks @xiangpengzhao , +1

**Comment by xiangpengzhao** [comment]
@m1093782566 awesome!

**Comment by frodenas** [comment]
@m1093782566 yep, this is a duplicate then. I checked the issue for PR and assignment references before working on my PR. I see you added them now, good. Closing this issue.

**Comment by m1093782566** [comment]
Thanks!

**Pull request closed without merging**